parser/proto.go

```diff
@@ -176,91 +240,99 @@ func (m *ProtoMetric) GetMap(key string, typeInfo *model.TypeInfo) interface{} {
 }
 
 func (m *ProtoMetric) GetIPv4(key string, nullable bool) interface{} {
-	value, _ := m.msg.TryGetFieldByName(key)
-	if value == nil {
+	field := m.msg.Descriptor().Fields().ByName(protoreflect.Name(key))
+	if field == nil {
 		if nullable {
 			return nil
 		}
 		return net.IPv4zero.String()
 	}
 
+	value := m.getValueOrList(field)
```

Comment:
I got you why you call `m.getValueOrList(field)` for `GetUUID`,  `GetIPv4`, `GetIPv6`. Because we have test cases with array for this methods. But I don't understand why these cases exist, ClickHouse doesn't allow inserting such values into a column with type UUID, IPv4, IPv6. These test cases are wrong.

Reply:
Please, see [my MR](https://github.com/viru-tech/clickhouse_sinker/pull/18) before making any changes. I've already resolved some threads in MR while I was figuring out the logic. If you agree, just merge it.